File: SchoolData/src/main/java/org/gb/homework/schooldata/controller/StudentController.java
package org.gb.homework.schooldata.controller;

import org.gb.homework.schooldata.model.Student;
import org.gb.homework.schooldata.service.UserService;
import org.gb.homework.schooldata.view.StudentView;

public class StudentController {
    private final UserService<Student> studentService;
    private final StudentView view;
    public Student currentStudent;

    public StudentController(UserService<Student> service, StudentView view) {
        this.studentService = service;
        this.view = view;
    }


    public void printAll() {
        var users = studentService.getAllUsers();
        view.showAll(users);
    }

    public void run() {
        var isWorking = true;

        while (isWorking) {

            var numberTask = view.showMenu(currentStudent);

            switch (numberTask) {
                case 0 -> isWorking = false;
                case 1 -> printAll();
                case 2 -> chooseStudent();
                case 3 -> addStudent();
                case 4 -> editStudent();
                case 5 -> deleteStudent();
                default -> view.showTaskError();
            }
        }

    }

    private void deleteStudent() {
        if (currentStudent == null) {
            view.showCurrentStudentError();
            return;
        }
        studentService.remove(currentStudent);
        currentStudent = null;
    }

    private void editStudent() {
        if (currentStudent == null) {
            view.showCurrentStudentError();
            return;
        }

        var name = view.inputName();
        var grade = view.inputGrade();
        var year = view.inputYear();
        currentStudent.setName(name);
        currentStudent.setGrade(grade);
        currentStudent.setYear(year);
        studentService.edit(currentStudent);
    }

    private void addStudent() {
        var name = view.inputName();
        var grade = view.inputGrade();
        var year = view.inputYear();

        var nextId = studentService.getNextId();

        studentService.add(new Student(nextId, name, (float) grade, year));
    }

    private void chooseStudent() {
        var nextId = studentService.getNextId();
        var teacherId = view.getStudentId(nextId);

        currentStudent = studentService.getAllUsers().stream().filter(x -> x.getId() == teacherId)
                .findFirst().orElse(null);
    }
}

File: csharp/SchoolData/Controllers/StudentController.cs
using Gb.Homework.SchoolData.Model;
using Gb.Homework.SchoolData.Service;
using Gb.Homework.SchoolData.View;

namespace Gb.Homework.SchoolData.Controllers
{

    public class StudentController
    {
        private UserService<Student> studentService;
        private StudentView view;
        public Student? currentStudent;

        public StudentController(UserService<Student> service, StudentView view)
        {
            this.studentService = service;
            this.view = view;
        }


        public void PrintAll()
        {
            var users = studentService.GetAllUsers();
            view.ShowAll(users);
        }

        public void Run()
        {
            var isWorking = true;

            while (isWorking)
            {

                var numberTask = view.ShowMenu(currentStudent);

                switch (numberTask)
                {
                    case 0:
                        isWorking = false;
                        break;
                    case 1:
                        PrintAll();
                        break;
                    case 2:
                        ChooseStudent();
                        break;
                    case 3:
                        AddStudent();
                        break;
                    case 4:
                        EditStudent();
                        break;
                    case 5:
                        DeleteStudent();
                        break;
                    default:
                        view.showTaskError();
                        break;
                }
            }

        }

        private void DeleteStudent()
        {
            if (currentStudent == null)
            {
                view.ShowCurrentStudentError();
                return;
            }
            studentService.Remove(currentStudent);
            currentStudent = null;
        }

        private void EditStudent()
        {
            if (currentStudent == null)
            {
                view.ShowCurrentStudentError();
                return;
            }

            var name = view.InputName();
            var grade = view.InputGrade();
            var year = view.InputYear();
            currentStudent.Name = name;
            currentStudent.Grade = grade;
            currentStudent.Year = year;
            studentService.Edit(currentStudent);
        }

        private void AddStudent()
        {
            var name = view.InputName();
            var grade = view.InputGrade();
            var year = view.InputYear();

            var nextId = studentService.GetNextId();

            studentService.Add(new Student(nextId, name, (float)grade, year));
        }

        private void ChooseStudent()
        {
            var nextId = studentService.GetNextId();
            var teacherId = view.getStudentId(nextId);

            currentStudent = studentService.GetAllUsers().Where(x => x.Id == teacherId).FirstOrDefault();
        }
    }

}
